src/pages/fi-pv-expenses/forms/PaymentVoucherExpensesForm.js

```diff
@@ -0,0 +1,732 @@
+          { from: 'name', to: 'etName' },
+          { from: 'reference', to: 'etRef' }
+        ],
+        columnsInDropDown: [
+          { key: 'reference', value: 'Reference' },
+          { key: 'name', value: 'Name' }
+        ],
+        readOnly: isPosted || isCancelled
+      }
+    },
+    {
+      component: 'textfield',
+      label: labels.expenseName,
+      name: 'etName',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+        component: 'textfield',
+        label: labels.subtotal,
+        name: 'subtotal',
+        props: {
+          readOnly: true
+        }
+    },
+    {
+        component: 'checkbox',
+        label: labels.isVAT,
+        name: 'isVAT',
+        props: {
+          disabled: isPosted || isCancelled
+        },
+        async onChange({ row: { update, newRow } }) {
+          if (newRow.isVAT && newRow.amount) {
+            let newSubtotal = newRow.amount * (100 / (100 + formik.values.vatPct));
+            update({
+              subtotal: newSubtotal.toFixed(2),
+              vatAmount: (newRow.amount - newSubtotal).toFixed(2)
+            })
+          } else {
+            update({
+              subtotal: newRow.amount,
+              vatAmount: 0
+            })
+          }
+        }
+    },
+    {
+        component: 'textfield',
+        label: labels.vat,
+        name: 'vatAmount',
+        props: {
+          readOnly: true
+        }
+    },
+    {
+        component: 'numberfield',
+        label: labels.amount,
+        name: 'amount',
+        props: {
+          readOnly: isPosted || isCancelled
+        },
+        async onChange({ row: { update, newRow } }) {
+          if (newRow.isVAT) {
+            let newSubtotal = newRow.amount * (100 / (100 + formik.values.vatPct));
+            update({
+              subtotal: newSubtotal.toFixed(2),
+              vatAmount: (newRow.amount - newSubtotal).toFixed(2)
+            })
+          } else {
+            update({
+              subtotal: newRow.amount,
+              vatAmount: 0
+            })
+          }
+        }
+    },
+    {
+        component: 'textfield',
+        label: labels.vendorName,
+        name: 'supplierName',
+        props: {
+          readOnly: isPosted || isCancelled
+        }
+    },
+    {
+      component: 'textfield',
+      label: labels.vatNo,
+      name: 'taxRef',
+      props: {
+        readOnly: isPosted || isCancelled
+      }
+  },
+  {
+      component: 'textfield',
+      label: labels.notes,
+      name: 'notes',
+      props: {
+        readOnly: isPosted || isCancelled
+      }
+  },
+  {
+      component: 'button',
+      name: 'hasCostCenters',
+      img: 'costCenters',
+      label: labels.costCenter,
+      onClick: (e, row, update, updateRow) => {
+        stack({
+          Component: ExpensesCostCenters,
+          props: {
+            maxAccess,
+            labels,
+            recordId,
+            row,
+            updateRow,
+            isPosted,
```

Comment:
remove isPosted , isCancelled and pass readOnly: isCancelled || isPosted

Reply:
I am using them in the DataGrid for cost centers 
```
 allowDelete={!isPosted && !isCancelled}
allowAddNewLine={!isPosted && !isCancelled}
```
for different condition